### PR TITLE
Initializing memory in RBC 

### PR DESCRIPTION
**Pull request by cjnolet**

### CODE REVIEW

File: cpp/include/raft/spatial/knn/detail/ball_cover.cuh [hunk 1]
```diff
@@ -340,6 +363,16 @@ void rbc_all_knn_query(const raft::handle_t& handle,
...
+               R_knn_inds.begin(),
+               R_knn_inds.end(),
+               std::numeric_limits<value_idx>::max());
+  thrust::fill(handle.get_thrust_policy(),
+               R_knn_dists.begin(),
+               R_knn_dists.end(),
+               std::numeric_limits<value_t>::max());
+
   // For debugging / verification. Remove before releasing
   rmm::device_uvector<value_int> dists_counter(index.m, handle.get_stream());
```
Comment (vinaydes) [inline cpp/include/raft/spatial/knn/detail/ball_cover.cuh:377]: While we are here, should this be removed? It seems, the `dists_counter` and `post_dists_counter` are not used by underline kernels.

Comment (cjnolet) [reply cpp/include/raft/spatial/knn/detail/ball_cover.cuh:377]: I've been using them for debugging (this algorithm is slow going and still isn't quite "finished").

**Comment by cjnolet** [comment]
rerun tests

**Review by dantegd — APPROVED** [review]

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**